subsys/net/ip/dhcpv4.c

```diff
@@ -159,6 +165,9 @@ static struct net_pkt *dhcpv4_create_message(struct net_if *iface, u8_t type,
 	size_t size = DHCPV4_MESSAGE_SIZE;
 	struct net_pkt *pkt;
 	struct dhcp_msg *msg;
+#if defined(CONFIG_NET_HOSTNAME_ENABLE)
+	const char *hostname = net_hostname_get();
+#endif
```

Comment:
I think we should calculate the string length only once here, and then use that value below. Now we call strlen() multiple times.

Reply:
Of course, thank you.